src/main/java/seedu/address/logic/commands/PermaSortContactCommand.java

```diff
@@ -0,0 +1,140 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Comparator;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+
+
+/**
+ * Sorts the contacts in Athena's address book permanently.
+ * Index entered determines the specific order.
+ */
+public class PermaSortContactCommand extends Command {
+
+    public static final String COMMAND_WORD = "psort";
+
+    public static final String COMMAND_TYPE = CommandType.CONTACT.toString();
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sort your contacts permanently "
+            + "by the index command entered  "
+            + "1 will be sort by alphabetical order of their names\n"
+            + "2 will be sort by alphabetical order of their address\n"
+            + "3 will be sort by alphabetical order of their first tag\n"
+            + "Parameters: INDEX (must be between 1 and 3) "
+            + "Example: " + COMMAND_WORD + " 1 ";
+
+    public static final String MESSAGE_INVALID_INDEX = "Invalid index entered, "
+            + "refer to below for the command's proper usage: "
+            + MESSAGE_USAGE;
+
+    private static final Comparator<Person> NAME_COMPARATOR = new Comparator<Person>() {
+        @Override
+        public int compare(Person o1, Person o2) {
+            return o1.getName().fullName.compareToIgnoreCase(o2.getName().fullName);
+        }
+    };
+
+    private static final Comparator<Person> ADDRESS_COMPARATOR = new Comparator<Person>() {
+        @Override
+        public int compare(Person o1, Person o2) {
+            return o1.getAddress().value.compareToIgnoreCase(o2.getAddress().value);
+        }
+    };
+
+    private static final Comparator<Person> TAG_COMPARATOR = new Comparator<Person>() {
+        @Override
+        public int compare(Person o1, Person o2) {
+            if (o1.getTags().size() == 0 && o2.getTags().size() != 0) {
+                return 1;
+            }
+            if (o1.getTags().size() != 0 && o2.getTags().size() == 0) {
+                return -1;
+            }
+            if (o1.getTags().size() == 0 && o2.getTags().size() == 0) {
+                return o1.getName().fullName.compareToIgnoreCase(o2.getName().fullName);
+            }
+            return o1.getTags().iterator().next().tagName
+                    .compareToIgnoreCase(o2.getTags().iterator().next().tagName);
+        }
+    };
```

Comment:
It seems to me that this comparator has some issues. In particular, the tags are actually stored in Java's `Set` which has no guarantee of order preservation. So this could result in different comparator results on different calls of the compare method.

Reply:
I believe this might be related to the same Comparator in SortContactCommand

Reply:
even if it doesn't maintain order preservation, this comparator will always compare the first tag in the set which is also the first tag shown on the contacts display. So it does fulfill what it needs to do?

Reply:
Is there a guarantee that the first tag being displayed in the GUI is the same as the first tag found in the `set.iterator()`?

On a related note, I wonder if this comparator is necessary because it seems quite arbitrary. I was going to suggest perhaps some sort of way to request sorting in the reverse order, so the name comparison would put Z at the front etc. In this case, I think a comparator that allows sorting by number of tags might fit this better. Though I am unsure how this might be useful as well.

Reply:
okay noted, will change it to comparing email so users can find all people with similar emails easily. For example, sorting by alphabetical order to find everyone who has the nus prefix email quickly

Reply:
Do reflect these changes within the origin SortContactCommand as well. Thanks!

Reply:
Yup, please refer to the permaSortCommand PR to see the changes made